### DISCUSSION TITLE
Expanded logging capabilities

### DISCUSSION THREAD

**Discussion by trevorstarick** [pocketbase/pocketbase#1960] [Show and tell]
I've been using Pocketbase as a Go app framework for handful of side projects and decided to see if I can come up with a proof of concept for "extra" logging capabilities: https://github.com/trevorstarick/pocketbase/tree/expanding-logs-func. 

It's heavily based on my logging framework of choice [rs/zerolog](https://github.com/rs/zerolog), as well as experience working with [grafana/loki](https://github.com/grafana/loki). It's currently rough around the edges, but I'd like to have a spot for an open discussion on how expanded logging could work, and am intending on spearheading this initiative. 

Some considerations that I think should be discussed is how best it's implementation can be done to match the feature-set that the most frequently used logging frameworks have, as well as the proposed [x/exp/slog](https://pkg.go.dev/golang.org/x/exp/slog) library. Additionally, DB I/O should be considered. I can imagine situation where log tables get large enough to cause performance issues, so should warnings be shown if the DB size gets to a certain "high-water mark" and should log rotation be a feature?

**Comment by ganigeorgiev** [comment]
Plans to "generalize" the logs and to enable custom and non-request related logs is in the [roadmap](https://github.com/orgs/pocketbase/projects/2) but for now remains a low priority.

I haven't thought much about it yet so I'm not sure what to suggest or recommend.

**Reply by zakkor** [reply]
@ganigeorgiev this is related to logging so I thought I'd leave this here instead of creating a new post:


Would it be possible to expand the current logging capabilities to include the **request form data** as well?

<img width="702" alt="image" src="https://github.com/pocketbase/pocketbase/assets/6191421/8bcb552b-2280-41f7-aeb2-31bfd38b8b09">

Just a new field here with the request data + maybe also the response data. Would be really useful because in a scenario like this, I can see an error happened but there's no way to debug it further
